includes/data-stores/class-wc-product-variation-data-store-cpt.php

```diff
@@ -179,32 +179,32 @@ public function update( &$product ) {
 
 	/**
 	 * Generates a title with attribute information for a variation.
-	 * Products with 2+ attributes with one-word values will get a title of the form "Name - Attribute: Value, Attribute: Value"
-	 * All other products will get a title of the form "Name - Value, Value"
+	 * Products will get a title of the form "Name - Value, Value" or just "Name".
 	 *
 	 * @since 3.0.0
 	 * @param WC_Product
 	 * @return string
 	 */
 	protected function generate_product_title( $product ) {
-		$include_attribute_names = false;
 		$attributes = (array) $product->get_attributes();
 
-		// Determine whether to include attribute names through counting the number of one-word attribute values.
-		$one_word_attributes = 0;
-		foreach ( $attributes as $name => $value ) {
-			if ( false === strpos( $value, '-' ) ) {
-				++$one_word_attributes;
-			}
-			if ( $one_word_attributes > 1 ) {
-				$include_attribute_names = true;
-				break;
+		// Don't include attributes if the product has 3+ attributes.
+		$should_include_attributes = count( $attributes ) < 3;
+
+		// Don't include attributes if an attribute name has 2+ words.
+		if ( $should_include_attributes ) {
+			foreach( $attributes as $name => $value ) {
+				if ( false !== strpos( $name, '-' ) ) {
```

Comment:
Should this check the label, not the slug with hyphens? The label gets displayed.